internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java

```diff
@@ -32,9 +33,18 @@
  */
 
 public class AvroStoreClientEndToEndTest extends AbstractClientEndToEndSetup {
-  protected void runTest(ClientConfig.ClientConfigBuilder clientConfigBuilder, boolean batchGet, int batchGetKeySize)
-      throws Exception {
-    runTest(clientConfigBuilder, batchGet, batchGetKeySize, (metricsRepository) -> {}, Optional.empty());
+  protected void runTest(
+      ClientConfig.ClientConfigBuilder clientConfigBuilder,
+      boolean batchGet,
+      int batchGetKeySize,
+      boolean useRouterBasedMetadata) throws Exception {
```

Comment:
NIT: Although we are using router to perform cluster discovery but the endpoint is on the server. Should we rename this to `useRequestBasedMetadata`?